src/schema/v2/artwork/index.ts

```diff
@@ -1451,13 +1489,51 @@ const Artwork: GraphQLFieldConfig<void, ResolverContext> = {
       description: "Include unlisted artwork or not",
     },
   },
-  resolve: (_source, args, { artworkLoader }) => {
+  resolve: async (
+    _source,
+    args,
+    { artworkLoader, marketPriceInsightsBatchLoader },
+    resolveInfo
+  ) => {
     const { id } = args
     const gravityParams = _.mapKeys(
       _.pick(args, ["includeUnlisted"]),
       (_v, k) => _.snakeCase(k)
     )
-    return artworkLoader(id, gravityParams)
+
+    const hasRequestedPriceInsights = isFieldRequested(
+      "marketPriceInsights",
+      resolveInfo
+    )
+
+    const gravityArtworkRes = await artworkLoader(id, gravityParams)
+
+    const enrichedArtwork = {
+      ...gravityArtworkRes,
+    }
+
+    // We don't want to query for the price insights unless the user has requested them
+    if (
+      marketPriceInsightsBatchLoader &&
+      gravityArtworkRes &&
+      hasRequestedPriceInsights
+    ) {
+      const marketPriceInsight = await marketPriceInsightsBatchLoader([
+        {
+          artistId: gravityArtworkRes.artist._id,
+          medium: gravityArtworkRes.medium,
+          category: gravityArtworkRes.category,
+        },
+      ])
+
+      if (marketPriceInsight.length > 0) {
+        enrichedArtwork.marketPriceInsights = marketPriceInsight[0]
+      } else {
+        enrichedArtwork.marketPriceInsights = null
+      }
+    }
```

Comment:
Can we use [enrichArtworksWithPriceInsights](https://github.com/artsy/metaphysics/blob/47c071848ab12900d5996d08d44faa33fae1ba67/src/schema/v2/me/myCollection.ts#L272)? Maybe we need to modify it but it would be nice to reuse the logic we already have.

Reply:
good idea